example/newreposecret/main.go

```diff
@@ -0,0 +1,197 @@
+*/
+func main() {
+	flag.Parse()
+
+	token := os.Getenv("GITHUB_AUTH_TOKEN")
+	if token == "" {
+		log.Fatal(`Unauthorized: No token present in environment variables. 
+Add your github api token as an environment variable named GITHUB_AUTH_TOKEN
+
+ex:
+	export GITHUB_AUTH_TOKEN=<github auth token>`)
+	}
+
+	if *repo == "" {
+		log.Fatal(`required flag repo was not passed. 
+please pass the repo you want to add this flag to as part of the command
+
+ex:
+	go run main.go --repo <repo name> --owner <repo owner name> 
+`)
+	}
+
+	if *owner == "" {
+		log.Fatal(`required flag owner was not passed. 
+please pass the owner you want to add this flag to as part of the command
+
+ex:
+	go run main.go --repo <repo name> --owner <repo owner name> 
+`)
+	}
+
+	ctx, client, err := GithubAuth()
+	if err != nil {
+		log.Fatal("unable to authorize client with github using token found in env var GITHUB_AUTH_TOKEN, error was passed: ", err.Error())
+	}
+	secretName, err := getSecretName()
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	secretValue, err := getSecretValue(secretName)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	addedSecretName, err := AddRepoSecret(ctx, client, *owner, *repo, secretName, secretValue)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	fmt.Println(fmt.Sprintf("Added Secret %s to the repo %s/%s", addedSecretName, *owner, *repo))
+
+}
+
+func getSecretName() (string, error) {
+	if len(os.Args) < 2 {
+		err := errors.New("no secrets passed in the go run command. pass in a secret name and then ensure to have the secret variable as an environment variable")
+		return "", err
+	}
+	args := os.Args
+	secret := args[3:]
+	return secret[0], nil
+}
+
+func getSecretValue(secretName string) (string, error) {
+	secretValue := os.Getenv(secretName)
+	if secretValue == "" {
+		return "", errors.New(fmt.Sprintf(`secret with name: %s not defined as an environment variable, 
+please export the variable as an enviornment variable so it can be read in
+
+ex:
+	export %s="secret value"
+`, secretName, secretName))
+	}
+	return secretValue, nil
+}
+
+// GithubAuth reads an api token from the environment
+// expecting API_GITHUB_TOKEN and authenticates with github api
+// and returns an authenticated client.
+func GithubAuth() (context.Context, *github.Client, error) {
+	token := os.Getenv("API_GITHUB_TOKEN")
+	if token == "" {
+		return nil, nil, errors.New(`no API_GITHUB_TOKEN was found in the environment variables.
+This is needed to authenticate with github. Please generate a github token that has access to do what you are trying to do
+and export it as an environment variable.
+
+ex:
+	export API_GITHUB_TOKEN="<token contents string>"
+`)
+	}
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+
+	client := github.NewClient(tc)
+	return ctx, client, nil
+}
+
+// AddRepoSecret will add a secret value to a given github repo for a given owner
```

Comment:
This comment is super-hard to read.  Please break it up into paragraphs with a blank comment line separating the paragraphs.
Also, please capitalize `GitHub` properly and make complete sentences ending in periods.

Reply:
yeah this was a nightmare after re-reading it. I rewrote it and spaced things out more which is hopefully more readable. It is all under the hood stuff too, so I mostly intended it to explain the process so that if people look at it they can hopefully understand why it has to be so complicated.